### PR TITLE
Badge and PresenceBadge: update to use `makeResetStyles`

### PR DESCRIPTION
**Pull request by spmonahan**
## Previous Behavior

Badges have many CSS classes on each slot.

## New Behavior

Badge and PresenceBadge have fewer classes on their slots. CounterBadge is left unchanged as its styles won't benefit from this change. Note this improvement will compound with similar changes to Avatar and Persona.

### Badge

Number of classes. In cases where there are multiple components in a story the first component is always used.

| Story | Old | New |
| --- | --- | --- |
| default | 49 | 5 |
| apperance | 49 | 5 | 
| sizes | 45 | 16 |
| shapes | 49 | 9 |
| color | 49 | 5 |
| icon | 62 | 11 |

### PresenceBadge

Number of classes. In cases where there are multiple components in a story the first component is always used.

| Story | Old | New | 
| --- | --- | --- |
| default | 22 | 9 |
| sizes | 26 | 13 |
| status | 22 | 9 |
| out of office | 22 | 9 |

## Related Issues

- #26811
- #26814

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/48b466e0ba8a55ad6f216127e538076c1e65fd37'>48b466e0ba8a55ad6f216127e538076c1e65fd37</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/Avatar-mount_base.html">1295</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/Avatar-mount.html">1205</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/Button-mount_base.html">965</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/Button-mount.html">963</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/Field-mount_base.html">1870</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/Field-mount.html">2043</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/FluentProvider-mount_base.html">1623</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/FluentProvider-mount.html">1634</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/FluentProviderWithTheme-mount_base.html">626</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/FluentProviderWithTheme-mount.html">555</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">592</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">596</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">632</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">568</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/InfoButton-mount_base.html">515</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/InfoButton-mount.html">581</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/MakeStyles-mount_base.html">1911</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/MakeStyles-mount.html">1899</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/Persona-mount_base.html">3111</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/Persona-mount.html">2724</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/SpinButton-mount_base.html">2590</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26817/perf-test-react-components/SpinButton-mount.html">2546</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-badge/src/components/Badge/useBadgeStyles.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { shorthands, mergeClasses, makeStyles } from '@griffel/react';
+import { shorthands, mergeClasses, makeResetStyles, makeStyles } from '@griffel/react';
```
Comment (khmakoto) [inline packages/react-components/react-badge/src/components/Badge/useBadgeStyles.ts:1]: _nit:_ Take this opportunity to order these alphabetically

```suggestion
import { makeResetStyles, makeStyles, mergeClasses, shorthands } from '@griffel/react';
```

File: packages/react-components/react-badge/src/components/PresenceBadge/usePresenceBadgeStyles.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { mergeClasses, makeStyles, shorthands } from '@griffel/react';
+import { mergeClasses, makeResetStyles, makeStyles } from '@griffel/react';
```
Comment (khmakoto) [inline packages/react-components/react-badge/src/components/PresenceBadge/usePresenceBadgeStyles.ts:1]: _nit:_ Take this opportunity to order these alphabetically

```suggestion
import { makeResetStyles, makeStyles, mergeClasses } from '@griffel/react';
```

File: packages/react-components/react-badge/src/components/Badge/useBadgeStyles.ts [hunk 3]
```diff
@@ -73,33 +94,23 @@ const useRootStyles = makeStyles({
     ...shorthands.borderRadius(tokens.borderRadiusSmall),
   },
   circular: {
-    ...shorthands.borderRadius(tokens.borderRadiusCircular),
+    // Set by useRootClassName
   },
 
   // border (all appearances except ghost)
```
Comment (khmakoto) [inline packages/react-components/react-badge/src/components/Badge/useBadgeStyles.ts:100]: This comment no longer applies, it should be changed to something like

`// Do not display the border pseudo-element when appearance is ghost.`

File: packages/react-components/react-badge/src/components/Badge/useBadgeStyles.ts [hunk 4]
```diff
@@ -12,16 +12,35 @@ export const badgeClassNames: SlotClassNames<BadgeSlots> = {
...
+  '::after': {
+    content: '""',
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    bottom: 0,
+    right: 0,
+    borderStyle: 'solid',
+    borderWidth: tokens.strokeWidthThin,
+    borderColor: 'inherit',
```
Comment (khmakoto) [inline packages/react-components/react-badge/src/components/Badge/useBadgeStyles.ts:38]: _nit:_ You can reduce this to one line:

```suggestion
    border: `${tokens.strokeWidthThin} solid inherit`,
```

**Review by khmakoto — APPROVED** [review]

File: change/@fluentui-react-badge-bd7ce6fd-4495-46a0-9d19-6aa724e86cc7.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-badge-bd7ce6fd-4495-46a0-9d19-6aa724e86cc7.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 19 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=26817&commitId=65e72531ffbdfbb7734521ba1c2e9744df59fc38&env=prod)

✅ There was 0 screenshots added, 0 screenshots removed, 1829 screenshots unchanged, 0 screenshots with different dimensions and 19 screenshots with visible difference.


  <details><summary><b>unknown</b> 19 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Badge Converged.filled - High Contrast.chromium.png|1398| Changed|
|Badge Converged.outline - Dark Mode.chromium.png|2780| Changed|
|Badge Converged.outline - High Contrast.chromium.png|5561| Changed|
|Badge Converged.outline.chromium.png|3153| Changed|
|Badge Converged.size- extra-large - RTL.chromium.png|1879| Changed|
|Badge Converged.size- extra-large.chromium.png|1879| Changed|
|Badge Converged.size- extra-small - RTL.chromium.png|69| Changed|
|Badge Converged.size- extra-small.chromium.png|69| Changed|
|Badge Converged.size- large - RTL.chromium.png|1536| Changed|
|Badge Converged.size- large.chromium.png|1536| Changed|
|Badge Converged.size- medium - RTL.chromium.png|1389| Changed|
|Badge Converged.size- medium.chromium.png|1389| Changed|
|Badge Converged.size- small - RTL.chromium.png|1139| Changed|
|Badge Converged.size- small.chromium.png|1139| Changed|
|Badge Converged.size- tiny - RTL.chromium.png|39| Changed|
|Badge Converged.size- tiny.chromium.png|39| Changed|
|Badge Converged.tint - Dark Mode.chromium.png|965| Changed|
|Badge Converged.tint - High Contrast.chromium.png|3283| Changed|
|Badge Converged.tint.chromium.png|488| Changed|

</details>


<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-badge/src/components/Badge/useBadgeStyles.ts (hunk 4 above)
Comment (spmonahan) [reply packages/react-components/react-badge/src/components/Badge/useBadgeStyles.ts]: I made that change and got a bunch of VR test failures. Turns out `inherit` cannot be used in CSS shorthands.

[From MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/Shorthand_properties):

> Only the individual properties values can inherit. As missing values are replaced by their initial value, it is impossible to allow inheritance of individual properties by omitting them. The keyword inherit can be applied to a property, but only as a whole, not as a keyword for one value or another. That means that the only way to make some specific value to be inherited is to use the longhand property with the keyword inherit.

[Codepen example](https://codepen.io/seanms/pen/wvEwOMY).

**Review by sopranopillow — APPROVED** [review]

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

**Pull request merged**